contrib/occtax/backend/occtax/migrations/35e6f0d01f93_restore_occtax_missing_counting.py

```diff
@@ -0,0 +1,303 @@
+"""restore occtax missing counting
+ 
+Revision ID: 35e6f0d01f93
+Revises: 944072911ff7
+Create Date: 2022-01-07 15:46:58.634720
+ 
+"""
+from alembic import op
+import sqlalchemy as sa
+ 
+ 
+# revision identifiers, used by Alembic.
+revision = '35e6f0d01f93'
+down_revision = '944072911ff7'
+branch_labels = None
+depends_on = None
+ 
+ 
+def upgrade():
+    op.execute("""
+        WITH
+            occ_id (id_occ) as (
+                SELECT DISTINCT o.id_occurrence_occtax
+                FROM pr_occtax.t_occurrences_occtax o
+                INNER JOIN gn_commons.t_history_actions h_create ON o.unique_id_occurence_occtax = h_create.uuid_attached_row AND h_create.operation_type = 'I'
+                INNER JOIN gn_commons.bib_tables_location bt ON bt.id_table_location = h_create.id_table_location AND bt.schema_name = 'pr_occtax' AND bt.table_name = 't_occurrences_occtax'
+                WHERE NOT EXISTS (SELECT NULL FROM pr_occtax.cor_counting_occtax c WHERE o.id_occurrence_occtax = c.id_occurrence_occtax)
+                --au besoin par rapport à la requête initiale, ajouter un filtre sur la date
+                --AND h_create.operation_date > '2022-01-12'
```

Comment:
Vu qu'il y a la jointure on ne mettrait par défaut la date de sortie de la 2.7
                AND h_create.operation_date > '2021-06-21' -- RELEASE 2.7

Reply:
Cool que la PR soit testée et OK chez vous. Idem pour la modification du trigger pour ne plus avoir les valeurs nulles quand on concatène ?
Pour la date, pas compris, avec la jointure il n'y a pas besoin de mettre la date directement ? Ou si il vaut la mettre ?

Reply:
La date est un garde fou permettant de ne pas remonter de données antérieures à l'apparition du bug. Il faudrait en théorie gérée autrement la correction de ces données.

La date reste indicative car cela dépend du moment ou la release a été déployée.

Reply:
OK on va donc inclure la date.
Et pour l'évolution du trigger, c'est en lien avec ce soucis : https://github.com/PnX-SI/GeoNature/issues/1467
OK pour toi ?

Par contre c'est bizarre, mais cette PR fait baisser la couverture du code par les tests. 🤨

Reply:
C'est ok pour la concaténation des champs additionnels. 
Je me pose juste une question sur le fait de les concaténer sans hiérarchie ce qui fait que si par hasard un champ additionel avec le même nom se retrouve à plusieurs niveaux (relevé, occurence), seule la première valeur sera concervée.
Est-ce qu'il ne faudrait pas prévoir une structure du type

```

Reply:
Ca corrige une partie de #1467 oui, mais il doit y avoir d'autre concaténation avec `||` ailleurs qui serait à trouver et reprendre.

Reply:
@amandine-sahl il n'y a plus de référence à relevé, occurrence, counting dans la synthèse. Ce doit être pour ça que ca avait été fait comme ça, en mode pot commun.

Reply:
OK donc pour la concaténation c'est mieux qu'avant. 
A voir par ailleurs si il y a d'autres endroits avec des soucis de concaténation similaire, et à voir par ailleurs pour améliorer la concaténation par niveau.

> Par contre la révision ne met à jour les données qu'en cas de problème de cd_nom.

Ça ne suffit pas ? Il peut y avoir des cas où les données dans la Synthèse ont été foirées par le soucis de dénombrement mais où le cd_nom n'est pas différents dans Occtax et dans Synthèse ?